packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -356,12 +356,10 @@ class _TextLayout {
     // Luckily they have the same bidi embedding level as the paragraph as per
     // https://unicode.org/reports/tr9/#L1, so we can anchor the caret to the
     // last logical trailing space.
-    final bool hasTrailingSpaces = switch (rawString.codeUnitAt(rawString.length - 1)) {
-      0x9 ||        // horizontal tab
-      0x3000 ||     // ideographic space
-      0x20 => true, // space
-      _ => false,
-    };
+    final bool hasTrailingSpaces = RegExp(
```

Comment:
could you link https://github.com/unicode-org/icu/blob/23d9628f88a2d0127c564ad98297061c36d3ce77/icu4c/source/common/unicode/uchar.h#L3388-L3425 in the comment for provenance?

Reply:
I believe this still includes every codepoint from `Zs`? Also I think we only care about the last code unit.

 Maybe:

```dart
final String lastCodeUnit = rawString[rawString.length - 1];
final bool hasTrailingSpaces = switch(lastCodeUnit.codeUnitAt(0)) { 
   0x0009 => true,
   0x00A0 || 0x2007 || 0x202F => false,
   _ => RegExp(r'p{Space_Separator}', unicode: true).hasMatch(lastCodeUnit),
};
```

Reply:
I updated code and comments at [953ec78](https://github.com/flutter/flutter/pull/152215/commits/953ec780ee50c62fd8b2081b1538ba23abb1a8d0).